libobs/obs-nix-platform.h

```diff
@@ -29,7 +29,7 @@ enum obs_nix_platform_type {
 #ifdef ENABLE_WAYLAND
 	OBS_NIX_PLATFORM_WAYLAND,
 #endif
-
+	OBS_NIX_PLATFORM_DRM,
 };
 
 /**
```

Comment:
To avoid API/ABI breakage `OBS_NIX_PLATFORM_WAYLAND` should be added even if Wayland is disabled.

Reply:
+1 we could leave wayland optional when it was at the end of the list, but now we just need it to be there always in the enum.

Reply:
Sure, it's just that at this point I spend precisely 0 seconds thinking about API, ABI, or any compatibility whatsoever. There's a plethora of reasons this PR is a draft :slightly_smiling_face: